src/main/java/net/masterthought/cucumber/ReportParser.java

```diff
@@ -81,6 +90,31 @@ public ReportParser(Configuration configuration) {
         return featureResults;
     }
 
+    private Feature[] suppressEmptyHooks(Feature[] features) {
+
+        for (Feature feature : features) {
+            Element[] elements = feature.getElements();
+            for (Element element : elements) {
+                Step[] steps = element.getSteps();
+                for (Step step : steps) {
+                    Hook[] after = step.getAfter();
+                    boolean isEmptyHook = false;
+                    for (Hook hook : after) {
+                        if (hook.getEmbeddings().length <= 0 && hook.getOutputs().length <= 0) {
+                            isEmptyHook = true;
+                            break;
+                        }
+                    }
+                    if (isEmptyHook) {
```

Comment:
I don't want to change the model as this may have interrupt other configuration. Instead I recommend to apply this condition in hooks.vm file